### ISSUE TITLE
Beacon/Validator Proposer Race Condition

### ISSUE THREAD

**Issue by mightypenguin** [sigp/lighthouse#1994]
## Description

Please provide a brief description of the issue.
Beacon syncs block right after (5 seconds?) validator asks for it.

## Version

Please provide your Lighthouse and Rust version. Are you building from
Pre-built binaries 1.0.0
(haven't had time to test against 1.0.1)

## Present Behaviour
I missed proposing a block on Pyrmont probably because of this.

## Expected Behaviour
If validator asks for a block that is missing it should retry some configurable period of time instead of failing completely.

## Steps to resolve
Add retries to validator (30 seconds? 1 minute).

Below are the logs for my lighthouse beacon and validator:

Beacon:
Nov 26 17:10:49.000 INFO Synced                                  slot: 59153, block: 0x8ed8…93e7, epoch: 1848, finalized_epoch: 1846, finalized_root: 0xb2fc…088c, peers: 55, service: slot_notifier
Nov 26 17:10:55.114 WARN Error processing HTTP API request       method: GET, path: /eth/v1/validator/blocks/59154, status: 500 Internal Server Error, elapsed: 5.3563ms
Nov 26 17:11:01.000 INFO Synced                                  slot: 59154, block:    …  empty, epoch: 1848, finalized_epoch: 1846, finalized_root: 0xb2fc…088c, peers: 55, service: slot_notifier
Nov 26 17:11:07.650 INFO New block received                      hash: 0x14eb…cf4d, slot: 59155

Validator:
Nov 26 17:07:13.001 INFO All validators active                   slot: 59135, epoch: 1847, total_validators: 2, active_validators: 2, proposers: 0, service: notifier
Nov 26 17:07:25.001 INFO All validators active                   slot: 59136, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:07:37.001 INFO All validators active                   slot: 59137, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:07:49.002 INFO All validators active                   slot: 59138, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:08:01.001 INFO All validators active                   slot: 59139, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:08:13.001 INFO All validators active                   slot: 59140, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:08:25.002 INFO All validators active                   slot: 59141, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:08:37.001 INFO All validators active                   slot: 59142, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:08:49.002 INFO All validators active                   slot: 59143, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:09:01.002 INFO All validators active                   slot: 59144, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:09:13.001 INFO All validators active                   slot: 59145, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:09:23.014 INFO Successfully published attestations     type: unaggregated, slot: 59146, committee_index: 22, head_block: 0x631921dd2ca69937dc2e558898c01b0059faed41e1fad6e84cac11954ea7ce2e, count: 1, service: attestation
Nov 26 17:09:25.001 INFO All validators active                   slot: 59146, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:09:37.002 INFO All validators active                   slot: 59147, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:09:49.002 INFO All validators active                   slot: 59148, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:10:01.002 INFO All validators active                   slot: 59149, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:10:13.002 INFO All validators active                   slot: 59150, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:10:25.002 INFO All validators active                   slot: 59151, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:10:37.001 INFO All validators active                   slot: 59152, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:10:47.014 INFO Successfully published attestations     type: unaggregated, slot: 59153, committee_index: 7, head_block: 0x8ed88a5a5ff60c58779e96f8667a2a92704db23584099e7f98f93b2784ae93e7, count: 1, service: attestation
Nov 26 17:10:49.001 INFO All validators active                   slot: 59153, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier
Nov 26 17:10:55.114 CRIT Error whilst producing block            message: Error from beacon node when producing block: ServerMessage(ErrorMessage { code: 500, message: "UNHANDLED_ERROR: NoEth1ChainConnection", stacktraces: [] }), service: block
Nov 26 17:11:01.002 INFO All validators active                   slot: 59154, epoch: 1848, total_validators: 2, active_validators: 2, proposers: 1, service: notifier

**Comment by michaelsproul** [comment]
It looks like your beacon node lacks an eth1 connection.

```
Nov 26 17:10:55.114 CRIT Error whilst producing block message: Error from beacon node when producing block: ServerMessage(ErrorMessage { code: 500, message: "UNHANDLED_ERROR: NoEth1ChainConnection", stacktraces: [] }), service: block
```

What are you using for eth1? A local node (it might have crashed)? Infura (it might have timed out)?

**Comment by mightypenguin** [comment]
Nethermind, but you can see the from the beacon log that it synchs the requested block 6 seconds after the Validator requests it.
It appears that the validator does not retry the request.
This was with Nethermind 1.0.0 so maybe it was fixed in 1.0.1 but I thought it worth looking into.

**Comment by paulhauner** [comment]
I'm closing this since this issue is inactive and I'm not sure where the issue is. I think there's some confusion between beacon blocks and eth1 blocks.
